x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java

```diff
@@ -251,12 +274,17 @@ static ShutdownShardMigrationStatus shardMigrationStatus(
                     );
                 }
             })
-            .map(Tuple::v1)
             .findFirst();
 
-        if (unmovableShard.isPresent()) {
+        if (totalRemainingShards == shardsToIgnoreForFinalStatus.get() && unmovableShard.isEmpty()) {
+            return new ShutdownShardMigrationStatus(
+                SingleNodeShutdownMetadata.Status.COMPLETE,
+                0,
+                "[" + shardsToIgnoreForFinalStatus.get() + "] primary shards cannot be moved away from this node"
```

Comment:
Might want a slightly more reassuring message here :)  (also, these aren't all just primary shards) perhaps something like:

`[3] shards cannot be moved away from this node but are replicated to another node in the cluster`

Reply:
Done, this wording is definitely better.